src/components/trigger-volume.js

```diff
@@ -0,0 +1,64 @@
+AFRAME.registerComponent("trigger-volume", {
+  schema: {
+    colliders: { type: "selectorAll" },
+    scope: { type: "selector" },
+    size: { type: "vec3", default: { x: 1, y: 1, z: 1 } },
+    target: { type: "string" },
+    enterComponent: { type: "string" },
+    enterProperty: { type: "string" },
+    enterValue: {
+      default: "",
+      parse: v => (typeof v === "object" ? v : JSON.parse(v)),
+      stringify: JSON.stringify
+    },
+    leaveComponent: { type: "string" },
+    leaveProperty: { type: "string" },
+    leaveValue: {
+      default: "",
+      parse: v => (typeof v === "object" ? v : JSON.parse(v)),
+      stringify: JSON.stringify
+    }
+  },
+  init() {
+    this.size = new THREE.Vector3();
+    this.boundingBoxWorldPosition = new THREE.Vector3();
```

Comment:
I like to only use instance properties for things that are actually carried over per frame and not just for these temp properties we create to avoid GC, for those I prefer using captured variables. This makes it easier to understand what is actually component state and has the added side benefit of sharing the temp variables across instances.

So I would keep `boundingBox` and `collidingLastFrame` as instance variables and make `boundingBox` and `colliderWorldPosition` into just `const`s outside the class (or if you like, captured in a closure on the function they are used in)